### DISCUSSION TITLE
Difference between available providers

### DISCUSSION THREAD

**Discussion by SolcAndMe** [wevm/wagmi#1067] [Q&A]
Why does `useProvider` return a different provider than `await (useAccount().connector.getSigner()).provider`? Is it because `useProvider` should not be dependent on a specific signer?
wagmi should expand the available provider hooks beyond `useWebSocketProvider` e.g. `useWeb3Provider`, `useJsonRpcProvider`.

**Comment by theodormarcu** [comment]
+1

This may be a useful answer: https://github.com/wagmi-dev/wagmi/discussions/1300
